### PR TITLE
Test on client & host scripts

### PR DESCRIPTION
**Pull request by shaopu1225**
- Test on client & host scripts
- Fix the primary object bug, set the first write object region to be the primary

### CODE REVIEW

File: s3-proxy/justfile [hunk 1]
```diff
@@ -37,7 +37,7 @@ run-registry:
     registry:2
 
 install-local-s3:
-  cargo install --force \
+  /home/ubuntu/.cargo/bin/cargo install --force \
```
Comment (lynnliu030) [inline s3-proxy/justfile:40]: Change it back

File: s3-proxy/src/skyproxy.rs [hunk 2]
```diff
@@ -141,7 +143,7 @@ impl SkyProxy {
             base_path: if local_server {
                 "http://127.0.0.1:3000".to_string()
             } else {
-                "http://18.102.61.249:3000".to_string()
+                "http://3.101.67.150:3000".to_string()
```
Comment (lynnliu030) [inline s3-proxy/src/skyproxy.rs:146]: Add a note here: #NOTE: ip address set to be the remote store-server addr

File: s3-proxy/src/skyproxy.rs [hunk 3]
```diff
@@ -666,54 +668,56 @@ impl S3 for SkyProxy {
...
-                                let put_resp = client.put_object(req).await.unwrap();
-                                let e_tag = put_resp.output.e_tag.unwrap();
-                                let head_resp = client
-                                    .head_object(S3Request::new(new_head_object_request(
-                                        locator.bucket.clone(),
-                                        locator.key.clone(),
-                                    )))
+                            .await;
+
+                            // only upload if start_upload is successful, this indicates that the object is not in the local object store
```
Comment (lynnliu030) [inline s3-proxy/src/skyproxy.rs:673]: Update the comment, this is not informative; this is dealing with the error of multi-concurrent pull-get?

File: s3-proxy/src/skyproxy_test.rs [hunk 4]
```diff
@@ -29,7 +29,7 @@ mod tests {
             CLIENT_FROM_REGION.clone(),
             true,
             true,
-            "push".to_string(),
+            "push".to_string(), // "copy_on_read"
```
Comment (lynnliu030) [inline s3-proxy/src/skyproxy_test.rs:32]: remove comment

File: skystore_cli.py [hunk 5]
```diff
@@ -120,7 +121,7 @@ def register(
     if local_test:
         server_addr = "localhost"
     else:
-        server_addr = "15.160.154.191"
+        server_addr = "3.101.67.150"
```
Comment (lynnliu030) [inline skystore_cli.py:124]: Add a note here as well

File: store-server/experiment/host.py [hunk 6]
```diff
@@ -177,8 +178,8 @@ def setup(server: compute.Server):
         # install cargo
         server.run_command(
             "curl https://sh.rustup.rs -sSf | sh -s -- -y; source $HOME/.cargo/env;\
-            git clone https://github.com/skyplane-project/skystore; cd skystore/store-server;\
-            pip3 install -r requirements.txt; cargo install just --force; \
+            git clone https://github.com/shaopu1225/skystore; cd skystore/store-server; git checkout skystore-main;\
```
Comment (lynnliu030) [inline store-server/experiment/host.py:181]: change it to the main repo and try again after merge

File: s3-proxy/src/skyproxy.rs (hunk 3 above)
Comment (shaopu1225) [reply s3-proxy/src/skyproxy.rs:673]: Yes.

**Pull request merged**